components/core/src/clp/GrepCore.hpp

```diff
@@ -162,15 +163,16 @@ std::optional<Query> GrepCore::process_raw_query(
     bool is_var;
     std::string search_string_for_sub_queries{search_string};
     if (use_heuristic) {
-        // Replace '?' wildcards with '*' wildcards since we currently have no support for
-        // generating sub-queries with '?' wildcards. The final wildcard match on the decompressed
-        // message uses the original wildcards, so correctness will be maintained.
-        std::replace(
-                search_string_for_sub_queries.begin(),
-                search_string_for_sub_queries.end(),
-                '?',
-                '*'
+        // Replace unescaped '?' wildcards with '*' wildcards since we currently have no support
+        // for generating sub-queries with '?' wildcards. The final wildcard match on the
+        // decompressed message uses the original wildcards, so correctness will be maintained.
```

Comment:
```suggestion
        // Replace unescaped '?' wildcards with '*' wildcards since we currently have no support for
        // generating sub-queries with '?' wildcards. The final wildcard match on the decompressed
        // message uses the original wildcards, so correctness will be maintained.
```